### ISSUE TITLE
[BUG] Open and Reopen of roll call fails

### ISSUE THREAD

**Issue by matteosz** [dedis/popstellar#1439]
### Description (Actual behavior)

Opening a roll call sometimes (I didn't find any specific pattern for which this happens) triggers the following exception:

> E/RollCallFragment: Could not open the rollcall
>     [An error response was replied by the server
>     Error -4 - invalid message field: failed to publish: failed to handle publish message: failed to process message: failed to process action 'roll_call#open': The field `opens` does not correspond to the id of the previous roll call message]
>     com.github.dedis.popstellar.utility.error.JsonRPCErrorException: Error -4 - invalid message field: failed to publish: failed to handle publish message: failed to process message: failed to process action **'roll_call#open'**: The field `opens` does not correspond to the id of the previous roll call message

As well as reopening a roll call. This happens with regularity when an user opens a roll call X, closes X, opens a roll call Y, closes Y, and then tries to reopen X. The exception is the following:

> E/RollCallFragment: Could not open the rollcall
> [An error response was replied by the server
> Error -4 - invalid message field: failed to publish: failed to handle publish message: failed to process message: failed to process action 'roll_call#reopen': The field `opens` does not correspond to the id of the previous roll call message]
> com.github.dedis.popstellar.utility.error.JsonRPCErrorException: Error -4 - invalid message field: failed to publish: failed to handle publish message: failed to process message: failed to process action **'roll_call#reopen'**: The field `opens` does not correspond to the id of the previous roll call message

### Expected behavior
The roll call should be reopened without throwing any exception.

### How to reproduce
In the following video there's an execution flow that shows the problem both on the opening and on the reopening (the screen glitches due to a bug with the recording).

[device-2023-03-07-143139.webm](https://user-images.githubusercontent.com/83547952/223437761-75051f42-0958-49c9-a8af-fd65c1ffbc06.webm)

### Version & Environment
This bug was reproduced on this [commit](https://github.com/dedis/popstellar/tree/bc23763f593c18dc39a0cf5446bc6352ac71582e), however it was present since this other [commit](https://github.com/dedis/popstellar/tree/7bf429b66dd8ed0c6d0bf2e8b461fa1bc254311d).

##### Front-ends:
- [ ] Fe1-Web (please include browser's names & version)
- [X] Fe2-Android (please specify if phone or emulation, and Android version)
- [ ] Not applicable

##### Back-ends:
- [ ] Be1-Go
- [ ] Be2-Scala
- [ ] Not applicable

##### Environment (as applicable):
 - Device: Pixel 6 API 30 as emulator
 
### Workaround
When it happens the user is forced to create a new roll call.

### Impact
Quite severe as the application still works but it's not anymore possible to either reopen or open a roll call.

**Comment by matteosz** [comment]
We tested with the Go server, need to see if the Scala one behaves like this as well

**Comment by pierluca** [comment]
What's the status on this ? @matteosz 

If I recall correctly, some issues were addressed on the front-end (considering the operations invalid in the first place), while others run deeper and required an actual fix.

If it makes sense to track the fixing more easily, feel free to split this issue into multiple issues addressing just one problem at a time.

**Comment by matteosz** [comment]
Yes, I put them in a single issues since the errors I was getting were almost identical, so perhaps there's a common problem at the root. 

Also I guess this is a backend related error, as this error is generated from the server response.
